src/common/utils.h

```diff
@@ -713,6 +713,20 @@ inline void EmplaceBackZeros(const NDArrayStorageType stype, const TShape &shape
   }
 }
 
+
+/*!
+ * \brief parallelize copy by OpenMP.
+ */
+template<typename DType>
+void OMPCopy(const TBlob &from, TBlob *to, const index_t size) {
+  DType* dst_dptr = to->dptr<DType>();
+  const DType* src_dptr = from.dptr<DType>();
+  #pragma omp parallel for num_threads(engine::OpenMP::Get()->GetRecommendedOMPThreadCount())
+  for (index_t i = 0; i < size; ++i) {
+    dst_dptr[i] = src_dptr[i];
```

Comment:
may exceed the boundary of src and dst? Do we really need parameter `size` here?

Reply:
yeh, you are right. we can fetch data size by `from` or `to`. We need to check if the sizes of `from` and `to` are equal before calling `OMPCopy`. So i don't want to re-fetch this parameter, just pass this parameter from `Copy` function in `ndarray/ndarray_function.cc`.